docs: refresh Filecoin Pin landing page and add Getting Started guide (#2459)
filecoin-project/filecoin-docs: builder-cookbook/filecoin-pin/getting-started.md

---
description: >-
  Install Filecoin Pin, connect your wallet, deposit storage credit, and pin
  your first file to Filecoin in around 10 minutes.
---

# Getting Started

This guide walks you through pinning your first file to Filecoin using the Filecoin Pin CLI. By the end, you'll be set up to:

* ✅ Install the Filecoin Pin CLI
* ✅ Connect your Ethereum-style wallet on Filecoin
* ✅ Deposit storage credit on Filecoin Pay

> * ✅ Pin a file to Filecoin and retrieve it via IPFS

Reviewer: ```suggestion
* ✅ Pin a file to Filecoin and retrieve it via IPFS standard tooling
```

***

## 🚀 Prerequisites

Before starting, make sure you have:

* **An Ethereum-style wallet on Filecoin** - MetaMask is the easiest option. If you don't have one set up, see [Wallets](../../basics/assets/wallets.md) and [Metamask setup](../../basics/assets/metamask-setup.md).
* **FIL in your wallet** - to pay transaction gas fees on Filecoin.
* **USDFC in your wallet** - to pay for storage. USDFC is the stablecoin used by Filecoin Onchain Cloud. The easiest way to get some is to [swap FIL for USDFC on Sushi](https://www.sushi.com/filecoin/swap?token0=NATIVE&token1=0x80b98d3aa09ffff255c3ba4a241111ff1262f045).
* **Node.js 22 or later** - the Filecoin Pin CLI runs on Node.js. Install from [nodejs.org](https://nodejs.org/) or via your package manager.

> <table data-view="cards"><thead><tr><th></th><th data-hidden></th><th data-hidden data-card-target data-type="content-ref"></th></tr></thead><tbody><tr><td>Filecoin Pin Repository</td><td><a href="https://github.com/filecoin-project/filecoin-pin">https://github.com/filecoin-project/filecoin-pin</a></td><td><a href="https://github.com/filecoin-project/filecoin-pin">https://github.com/filecoin-project/filecoin-pin</a></td></tr><tr><td>Filecoin Pin Support Channels</td><td><a href="https://filecoinproject.slack.com/archives/C07CGTXHHT4">Filecoin Slack - #fil-foc</a></td><td><a href="https://filecoinproject.slack.com/archives/C07CGTXHHT4">https://filecoinproject.slack.com/archives/C07CGTXHHT4</a></td></tr></tbody></table>

Reviewer: nit: why are all the URLs dupllicated? I'm surprised to see 4 `<a>` tags.

{% hint style="info" %}
**Coming from Storacha?** Refer to the email we sent you for the specific USDFC amount we'd recommend depositing for your dataset size.
{% endhint %}

#### 4️⃣ Confirm the deposit

The CLI submits the deposit transaction onchain and shows the transaction hash and your new storage capacity.

{% hint style="success" %}
You should see something like:

```
✓ Deposit complete
  Deposit tx: 0x1234...abcd

New Storage Capacity:
  Total deposit: 10.00 USDFC
  Capacity: ~XX GiB for 1 month
```
{% endhint %}

***

## 💵 Top up your runway (optional)

If you want to ensure your deposit covers a specific number of days at your current storage usage, run:

```sh
filecoin-pin payments fund --days 30
```

This deposits (or withdraws) the right amount of USDFC to give you exactly 30 days of runway based on what you're currently storing. You can use any number of days you like.

To check your current deposit balance, runway, and payment status at any time:

```sh
filecoin-pin payments status
```

***

## 📌 Pin your first file

Now you're ready to pin. Create a test file:

```sh

> echo "Hello Filecoin Pin!" > demo.txt

Reviewer: It would good to make this a unique string so that things don't "just work" because of content addressing.

Basically, lets do something like "Hello Filecoin Pin @ `date`!"

```

Pin it to Filecoin:

```sh
filecoin-pin add demo.txt
```

The CLI will:

1. Pack your file into IPFS-compatible format (a CAR file).
2. Select storage providers automatically.

> 3. Upload your file to two providers for redundancy.

Reviewer: ```suggestion
3. Store your file with two providers for redundancy.
```

Switched from upload since we actually only upload once.

> 4. Verify the upload was advertised on the IPFS network indexer.

Reviewer: ```suggestion
4. Verify the upload was advertised to [IPNI indexers](https://github.com/filecoin-project/filecoin-pin/blob/master/documentation/glossary.md#ipni).
```

{% hint style="success" %}
You should see something like:

```
✓ File validated (20 B)
✓ Connected to Filecoin Mainnet
✓ File packed with root CID: bafybeihkoviema7g3gxyt6la7vd5ho32ictqbilu3wnlo3rs7ewhnp7lly
✓ IPFS content loaded (96 B)

━━━ Add Complete ━━━

Root CID:  bafybeihkoviema7g3gxyt6la7vd5ho32ictqbilu3wnlo3rs7ewhnp7lly
Piece CID: bafkzcibcfab4grpgq6e6rva4kfuxfcvibdzx3kn2jdw6q3zqgwt5cou7j6k4wfq
Copies:    2/2

Add completed successfully
```
{% endhint %}

The **Root CID** is your IPFS Content Identifier - it's how you'll retrieve your data. Save it somewhere.

You can also pin a directory by passing the directory path instead of a file:

```sh
mkdir my-data
echo "File 1" > my-data/file1.txt
echo "File 2" > my-data/file2.txt
filecoin-pin add my-data/
```

***

## 🌐 Retrieve your file

> Your file is now retrievable via any IPFS Mainnet gateway using its Root CID. For example:

Reviewer: ```suggestion
Your file is now retrievable via standard IPFS tooling using its Root CID. For example:
```

Reviewer 2: Is this the best practice? To just use an ipfs gateway? I'm building a video player web app and was planning on using synapse, but am having issues. https://github.com/Haven-hvn/haven-dapp Would appreciate some guidance, on what you intend devs to use for retrievability.

Reviewer: @HavenCTO : This is getting into the general discussion of IPFS retrieval and generally using a common-good IPFS gateway isn't the "best practice". It works for proof of concept, but shouldn't be relied upon. 

Your question is making me realize that the FOC and filecoin-pin docs don't talk enough about retrieval. There should be a link I could send you that would answer this. https://docs.ipfs.tech/how-to/ipfs-in-web-apps/#retrieval has some info from an IPFS and web regard, but I think we should answer this holistically for an FOC regard. I have created https://github.com/FilOzone/synapse-sdk/issues/790 for this.

Before that gets implemented, feel free to discuss this in the #fil-foc Filecoin Slack channel.

```
https://<YOUR_ROOT_CID>.ipfs.inbrowser.link
```

Or via the dweb.link gateway:

```
https://dweb.link/ipfs/<YOUR_ROOT_CID>
```

Try opening one of those URLs in your browser - your file should load.

You can also retrieve it programmatically using any IPFS-compatible client (Kubo, Helia, Lassie, etc.) by referencing the Root CID.

***

## 🛡️ Inspect your storage proofs

> Filecoin storage providers must cryptographically prove daily that they continue to store your data. You can inspect those proofs and the on-chain payment rails at any time.

Reviewer: ```suggestion
Filecoin storage providers must cryptographically prove multiple times per day that they continue to store your data. You can inspect those proofs and the on-chain payment rails at any time.
```

You'll see something like:

```
Data Set #279 • live
  Managed by Warm Storage: yes
  Pieces stored: 2
  Total size: 672.0 B
  PDP rail ID: 631
  Payer: 0xYOUR_WALLET_ADDRESS
  Payee: 0xPROVIDER_ADDRESS

Provider
  Provider: <provider-name> (ID <N>)

Provider Service
  Service URL: https://<provider-service-url>
  Min proving period: 30 epochs

Pieces
  #0
    CommP: bafkzcib...
    Root CID: bafybei...
```
{% endhint %}

Key things to look for:

* **Status: `live`** - your data set is active and being proved.
* **PDP rail ID** - your active storage-proof payment rail.
* **Min proving period** - how often the provider must submit a fresh proof.
* **Provider Service URL** - direct retrieval endpoint for your pieces.
* **CommP / Root CID** per piece - the piece CID is what the provider proves; the Root CID is what you use to retrieve.

***

## 🎉 You're Done!

You've successfully pinned your first file to Filecoin. You now have:

* ✅ The Filecoin Pin CLI installed and configured
* ✅ Your wallet connected with funded payments on Filecoin Pay
* ✅ Files pinned to two Filecoin storage providers with daily cryptographic proofs
* ✅ Your content retrievable via standard IPFS gateways

***

> ## 🔜 Next Steps

Reviewer: Lets add a pointer to https://github.com/filecoin-project/filecoin-pin/blob/master/documentation/behind-the-scenes-of-adding-a-file.md so someone can learn more about what happening behind the scenes if they want to.
